coffee-chats/src/main/webapp/src/util/chatRequest.ts

```diff
@@ -49,3 +51,44 @@ export const submitChatRequest = async (interests: string[],
     return false;
   }
 };
+
+/**
+ * Fetches pending requests from backend for the current user, updates state with the response via 
+ * given function.
+ * 
+ * @param setPendingRequests - Function to set the current pending requests user has.
+ */
+export const fetchPendingRequests = (setPendingRequests: (requests: ChatRequest[]) => void) => {
+  fetch(getRequestsURL('pending'))
```

Comment:
Why not just use async functions?

Reply:
They are pretty much identical in behaviour, it just looked clearer here using fetch and then to show the data transformation from response to actual required objects.